src/transformers/models/intern_s1/tokenization_intern_s1.py

```diff
@@ -0,0 +1,981 @@
+            The end of sequence token.
+        pad_token (`str`, *optional*, defaults to `"<|endoftext|>"`):
+            The token used for padding, for example when batching sequences of different lengths.
+        clean_up_tokenization_spaces (`bool`, *optional*, defaults to `False`):
+            Whether or not the model should cleanup the spaces that were added when splitting the input text during the
+            tokenization process. Not applicable to this tokenizer, since tokenization does not add spaces.
+        split_special_tokens (`bool`, *optional*, defaults to `False`):
+            Whether or not the special tokens should be split during the tokenization process. The default behavior is
+            to not split special tokens. This means that if `<|endoftext|>` is the `eos_token`, then `tokenizer.tokenize("<|endoftext|>") =
+            ['<|endoftext|>`]. Otherwise, if `split_special_tokens=True`, then `tokenizer.tokenize("<|endoftext|>")` will be give `['<',
+            '|', 'endo', 'ft', 'ext', '|', '>']`. This argument is only supported for `slow` tokenizers for the moment.
+    """
+
+    vocab_files_names = VOCAB_FILES_NAMES
+    model_input_names = ["input_ids", "attention_mask"]
+
+    def __init__(
+        self,
+        vocab_file,
+        merges_file,
+        errors="replace",
+        unk_token="<|endoftext|>",
+        bos_token=None,
+        eos_token="<|endoftext|>",
+        pad_token="<|endoftext|>",
+        clean_up_tokenization_spaces=False,
+        split_special_tokens=False,
+        **kwargs,
+    ):
+        self.extra_tokenizer_start_mapping = {}
+        self.extra_tokenizer_end_mapping = {}
+        self._extra_special_tokens = []
+
+        self._extra_tokenizer_list = [
+            {
+                "tokenizer_name": "tokenizer_SMILES",
+                "tokenizer_path": os.path.join(os.path.dirname(vocab_file), "tokenizer_SMILES.model"),
+                "begin_sp_tokens": ["<SMILES>", "<SELFIES>"],
+                "end_sp_tokens": ["</SMILES>", "</SELFIES>"],
+                "auto_begin_sp_tokens": ["<SMILES_AUTO_DETECT>"],
+                "auto_end_sp_tokens": ["</SMILES_AUTO_DETECT>"],
+            },
+            {
+                "tokenizer_name": "tokenizer_IUPAC",
+                "tokenizer_path": os.path.join(os.path.dirname(vocab_file), "tokenizer_IUPAC.model"),
+                "begin_sp_tokens": ["<IUPAC>"],
+                "end_sp_tokens": ["</IUPAC>"],
+                "auto_begin_sp_tokens": [],
+                "auto_end_sp_tokens": [],
+            },
+            {
+                "tokenizer_name": "tokenizer_FASTA",
+                "tokenizer_path": os.path.join(os.path.dirname(vocab_file), "tokenizer_FASTA.model"),
+                "begin_sp_tokens": ["<FASTA>"],
+                "end_sp_tokens": ["</FASTA>"],
+                "auto_begin_sp_tokens": ["<FASTA_AUTO_DETECT>"],
+                "auto_end_sp_tokens": ["</FASTA_AUTO_DETECT>"],
+            },
```

Comment:
this is interesting, but I am wondering what is the use case? the user has to manually set `begin_sp_tokens` or `end_sp_tokens`? This is mostly used for encoding right? So why not simply automatically split the text, and prefix it with something -> make use of the specific tokens?